cmd/hub/mcscontroller/main.go

```diff
@@ -0,0 +1,135 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package main implements a controller for Kubernetes Multi-cluster
+// Services API as defined by KEP 1645.
+// The controller is responsible for coordinating a common view of exports
+// and imports on a shared (a.k.a. hub) cluster, and replaces the, per-workload-
+// cluster mapping, currently provided by Submariner's lighthouse component.
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/healthz"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+
+	mcsv1a1 "sigs.k8s.io/mcs-api/pkg/apis/v1alpha1"
+	// +kubebuilder:scaffold:imports
+)
+
+var (
+	scheme   = runtime.NewScheme()
+	setupLog = ctrl.Log.WithName("setup")
+)
+
+func init() {
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+	utilruntime.Must(mcsv1a1.AddToScheme(scheme))
+	//+kubebuilder:scaffold:scheme
+}
+
+func main() {
```

Comment:
the agent also has an uninstall option, I think you will need to support it as well.

Reply:
Not sure. Wouldn't kubcetl delete handle it for us? We are not creating additional cluster state that needs to be "installed" and "uninstalled"

Reply:
We create objects and we want to delete them when uninstalling. 
subctl has an "uninstall" option that executes a custom uninstallation procedure in all of its components. 
If you look at the agent's main.go you see that it supports it, so I think  the hub's main shall too.

Reply:
It may be added in a separate PR, just we may want to track it in an issue

Reply:
https://github.com/axon-net/lighthouse/issues/19